From 9c58af7528fb62825ab2998feab3e7244c833fce Mon Sep 17 00:00:00 2001
From: Mingliang Liu <liuml07@apache.org>
Date: Tue, 28 Feb 2017 13:22:27 -0800
Subject: [PATCH 2332/2848] HADOOP-14129. ITestS3ACredentialsInURL sometimes
 fails. Contributed by Sean Mackrory

(cherry picked from commit 32e9a24f905fb766369100e8a78f14434d7c6180)

Change-Id: I026883635cfbbb4d12078e562e6a6cff9d334df2
---
 .../hadoop/fs/s3a/ITestS3ACredentialsInURL.java    |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3ACredentialsInURL.java b/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3ACredentialsInURL.java
index f4f7fbb..5a4b2fc 100644
--- a/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3ACredentialsInURL.java
+++ b/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3ACredentialsInURL.java
@@ -84,6 +84,11 @@ public void testInstantiateFromURL() throws Throwable {
     conf.unset(Constants.ACCESS_KEY);
     conf.unset(Constants.SECRET_KEY);
     fs = S3ATestUtils.createTestFileSystem(conf);
+
+    // Skip in the case of S3Guard with DynamoDB because it cannot get
+    // credentials for its own use if they're only in S3 URLs
+    Assume.assumeFalse(fs.hasMetadataStore());
+
     String fsURI = fs.getUri().toString();
     assertFalse("FS URI contains a @ symbol", fsURI.contains("@"));
     assertFalse("FS URI contains a % symbol", fsURI.contains("%"));
-- 
1.7.9.5

